setup.sh

```diff
@@ -0,0 +1,268 @@
+#!/bin/bash
+
+ARCH=$(uname -m)
+LATEST_RELEASE_TAG=$(curl -s "https://api.github.com/repos/paradedb/paradedb/releases/latest" | jq -r .tag_name)
+LATEST_RELEASE_VERSION="${LATEST_RELEASE_TAG#v}"
+OSTYPE="darwin"
+
+EXIT_MSG="\n\nIf you'd like to stay upto date with everything about ParadeDB\nJoin our slack channel: https://join.slack.com/t/paradedbcommunity/shared_invite/zt-217mordsh-ielS6BiZf7VW3rqKBFgAlQ \nGitHub: https://github.com/paradedb/paradedb"
+
+set -Eeuo pipefail
+
+installDockerDepsLinux(){
+  echo "Which type of distro are you using?(Required to install dependencies)"
+  OPTIONS=("Debian Based" "RHEL Based" "Arch Based")
+
+  select opt in "${OPTIONS[@]}"
+  do
+    case $opt in
+      "Debian Base")
+        sudo apt-get install docker -y || false
+        break ;;
+      "RHEL Based")
+        sudo dnf install docker -y || false
+        break ;;
+      "Arch Based")
+        sudo pacman -Su docker || false
+        break ;;
+      *)
+        break ;;
+    esac
+  done
+
+}
+
+installDocker() {
```

Comment:
We should first check if it's installed. On macOS it's common to have it installed via Docker desktop and not `brew`. I have Docker installed, but it was going to do a duplicate install via Homebrew when I tested the script.

Claude has suggested something like this:

```bash
#!/bin/bash

# Function to check if a command exists
command_exists() {
    command -v "$1" >/dev/null 2>&1
}

# Check if Docker command is available
if command_exists docker; then
    echo "Docker command is available."
    
    # Get Docker version
    docker_version=$(docker --version)
    echo "Docker version: $docker_version"
    
    # Check if Docker daemon is running
    if docker info >/dev/null 2>&1; then
        echo "Docker daemon is running."
        
        # Check if it's Docker Desktop
        if docker info | grep -q "Operating System: Docker Desktop"; then
            echo "Docker is installed via Docker Desktop on macOS."
        else
            echo "Docker is installed, but it might not be Docker Desktop."
        fi
    else
        echo "Docker daemon is not running."
    fi
else
    echo "Docker is not installed or not in PATH."
fi
```

Reply:
cool I'll add this

Reply:
for mac os should we just prompt the user to install docker themselves?

Reply:
Hmm Perhaps yeah... Let's do that

Reply:
okk, all done